pkg/apiserver/resthandler.go

```diff
@@ -684,15 +686,29 @@ func DeleteResource(r rest.GracefulDeleter, checkBody bool, scope RequestScope,
 // to use it.
 // TODO: add appropriate structured error responses
 func queryToObject(query url.Values, scope RequestScope, kind string) (runtime.Object, error) {
```

Comment:
I'd rather see us remove ServerGroupVersion and follow up by cleaning up the "v1" fallback here than expanding the use of ServerGroupVersion like #17163 is